navigation/libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationView.java

```diff
@@ -151,28 +167,110 @@ public void userOffRoute(Location location) {
+      public void onClick(View view) {
+        sheetShadow.setVisibility(View.GONE);
+        summaryOptions.setVisibility(View.GONE);
+        summaryDirections.setVisibility(View.VISIBLE);
+      }
+    });
+    cancelBtn.setOnClickListener(new View.OnClickListener() {
+      @Override
+      public void onClick(View view) {
+        // TODO Finish with cancelled result code
+        finish();
+      }
+    });
+    expandArrow.setOnClickListener(new View.OnClickListener() {
+      @Override
+      public void onClick(View view) {
+        if (summaryBehavior.getState() == BottomSheetBehavior.STATE_COLLAPSED) {
+          summaryBehavior.setState(BottomSheetBehavior.STATE_EXPANDED);
+        } else {
+          summaryBehavior.setState(BottomSheetBehavior.STATE_COLLAPSED);
+        }
+      }
+    });
   }
 
   private void initMap(Bundle savedInstanceState) {
     mapView.onCreate(savedInstanceState);
     mapView.getMapAsync(this);
   }
 
+  private void initSummaryBottomSheet() {
+    summaryBehavior = BottomSheetBehavior.from(summaryBottomSheet);
+    summaryBehavior.setState(BottomSheetBehavior.STATE_COLLAPSED);
+    summaryBehavior.setBottomSheetCallback(new BottomSheetBehavior.BottomSheetCallback() {
+      @Override
+      public void onStateChanged(@NonNull View bottomSheet, int newState) {
+        switch (newState) {
+          case BottomSheetBehavior.STATE_EXPANDED:
+            cancelBtn.setClickable(false);
+            if (summaryDirections.getVisibility() == View.VISIBLE) {
+              sheetShadow.setVisibility(View.GONE);
+            }
+            break;
+          case BottomSheetBehavior.STATE_COLLAPSED:
+            cancelBtn.setClickable(true);
+            summaryOptions.setVisibility(View.VISIBLE);
+            summaryDirections.setVisibility(View.GONE);
+            break;
+          default:
+            break;
+        }
+      }
+
+      @Override
+      public void onSlide(@NonNull View bottomSheet, float slideOffset) {
+        if (slideOffset < 1f && sheetShadow.getVisibility() != View.VISIBLE) {
+          sheetShadow.setVisibility(View.VISIBLE);
+        }
+        if (summaryDirections.getVisibility() == View.VISIBLE) {
+          instructionView.animate().alpha(1 - slideOffset).setDuration(0).start();
+        }
+        cancelBtn.animate().alpha(1 - slideOffset).setDuration(0).start();
+        expandArrow.animate().rotation(180 * slideOffset).setDuration(0).start();
+      }
+    });
+  }
+
   private void initNavigation() {
     navigation = new MapboxNavigation(this);
     navigation.addProgressChangeListener(this);
-
+    navigation.addProgressChangeListener(instructionView);
+    navigation.addProgressChangeListener(summaryBottomSheet);
+    navigation.addMilestoneEventListener(instructionView);
+    navigation.addOffRouteListener(this);
+    navigation.addOffRouteListener(summaryBottomSheet);
+    navigation.addOffRouteListener(instructionView);
   }
 
   @SuppressWarnings({"MissingPermission"})
```

Comment:
Does your activity handle permissions correctly or is this up to the developer?

Reply:
Up to the developer for now